### PR TITLE
Use Promises

### PR DESCRIPTION
**Pull request by TimothyGu**
Promises greatly simplify the callback hell, and make the code a lot easier to read.

### CODE REVIEW

File: lib/db.js [hunk 1]
```diff
@@ -1,29 +1,15 @@
+var Promise = require('bluebird');
 var levelup = require('level');
-var db = levelup('./data');
+var db = Promise.promisifyAll(levelup('./data'));
 
 exports.scores = {
 	get: function(id, cb) {
```
Comment (trevordixon) [inline lib/db.js]: Don't need `cb` argument.

**Comment by TimothyGu** [comment]
@trevordixon fixed.

Note that right now the diffstat shows that there are more additions than deletions. This is because of some refactoring and lone line-breaking in 65f3ea1.

**Comment by trevordixon** [comment]
Looks good. Merge whenever you'd like.

**Comment by TimothyGu** [comment]
Done. Thanks!

**Pull request merged**